### PR TITLE
[Feature] Department Form Backend

### LINKED ISSUE #7841: ✨ Directive forms - Department-specific recruitment process form - Backend
## ✨ Feature

Build the backend data model 

## 🕵️ Details

This is for "Department-specific recruitment process form"
Last backend one for reference #7273

## 🎨 Design File

Figma link to form design

https://www.figma.com/file/QPLlxobrg2ZQCW5nf1HDFx/Managers---Directive-forms?type=design&node-id=55-2241&mode=design

### 🧑‍🎨 Designer

Josh

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

Follow in the steps of #7273

## ✅ Acceptance Criteria

- [ ] Mutation to build the database tables
- [ ] Eloquent models and relationships
- [ ] GraphQL types, queries, and mutations to input and read out data
- [ ] same schema file as other directive form
- [ ] only `platform_admin` can query results (or submit them, for now)
- [ ] Backend validation

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #7841

## 👋 Introduction

This branch adds the backend work to create, update, and delete the Department Specific Recruitment Process Form.

## 🕵️ Details

- There's no frontend for this.  Use Graphiql to interact with the backend.

### Process

1. Reseed the roles and permissions
2. Refresh the GraphQL schema
3. Check that everything works:
    - [ ] Factory and seeder works
    - [ ] PHPUnit tests work
    - [ ] Can create a questionnaire
    - [ ] Can query questionnaires
    - [ ] Can update a questionnaire
    - [ ] Can delete a questionnaire
    - [ ] Validation catches basic errors
    - [ ] Policy enforces only platform admin usage

### Queries to get you started

<details>
  <summary>CreateDigitalContractingQuestionnaire</summary>
  
  ```graphql

mutation CreateDepartmentSpecificRecruitmentProcessForm($form: DepartmentSpecificRecruitmentProcessFormInput!) {
  createDepartmentSpecificRecruitmentProcessForm(
    departmentSpecificRecruitmentProcessForm: $form
  ) {
    id
    department {
      id
    }
    departmentOther
    recruitmentProcessLeadName
    recruitmentProcessLeadJobTitle
    recruitmentProcessLeadEmail
    postingDate
    positions {
      id
      classificationGroup
      classificationLevel
      jobTitle
      employmentTypes
      employmentTypesOther
    }
    advertisementType
    advertisingPlatforms
    advertisingPlatformsOther
    jobAdvertisementLink
  }
}
```

```json
{
  "form": {
    "department": {
      "connect": "ec817e0d-71ac-4481-9470-d2172f1c58cc"
    },
    "departmentOther": "",
    "recruitmentProcessLeadName": "recruitment_process_lead_name",
    "recruitmentProcessLeadJobTitle": "recruitment_process_lead_job_title",
    "recruitmentProcessLeadEmail": "recruitment@example.org",
    "postingDate": "2025-01-01",
    "positions": {
      "create": [
        {
          "classificationGroup": "classification_group",
          "classificationLevel": "classification_level",
          "jobTitle": "job_title",
          "employmentTypes": [
            "ASSIGNMENT",
            "OTHER"
          ],
          "employmentTypesOther": "employment_type_other"
        }
      ]
    },
    "advertisementType": "INTERNAL",
    "advertisingPlatforms": [
      "LINKEDIN",
      "OTHER"
    ],
    "advertisingPlatformsOther": "advertising_platforms_other",
    "jobAdvertisementLink": "google.com"
  }
}
  ```
</details>

<details>
  <summary>DepartmentSpecificRecruitmentProcessForms</summary>
  
  ```graphql
query DepartmentSpecificRecruitmentProcessForms {
  departmentSpecificRecruitmentProcessForms {
    id
    department {
      id
    }
    departmentOther
    recruitmentProcessLeadName
    recruitmentProcessLeadJobTitle
    recruitmentProcessLeadEmail
    postingDate
    positions {
      id
      classificationGroup
      classificationLevel
      jobTitle
      employmentTypes
      employmentTypesOther
    }
    advertisementType
    advertisingPlatforms
    advertisingPlatformsOther
    jobAdvertisementLink
  }
}
  ```
</details>

<details>
  <summary>DepartmentSpecificRecruitmentProcessForm</summary>
  
  ```graphql
query DepartmentSpecificRecruitmentProcessForm {
  departmentSpecificRecruitmentProcessForm(
    id: "20663acd-9571-4b26-a0dc-c0062f04ae86"
  ) {
    id
    department {
      id
    }
    departmentOther
    recruitmentProcessLeadName
    recruitmentProcessLeadJobTitle
    recruitmentProcessLeadEmail
    postingDate
    positions {
      id
      classificationGroup
      classificationLevel
      jobTitle
      employmentTypes
      employmentTypesOther
    }
    advertisementType
    advertisingPlatforms
    advertisingPlatformsOther
    jobAdvertisementLink
  }
}
  ```
</details>

<details>
  <summary>UpdateDigitalContractingQuestionnaire</summary>
  
  ```graphql

mutation updateDepartmentSpecificRecruitmentProcessForm($id: UUID!, $form: DepartmentSpecificRecruitmentProcessFormInput!) {
  updateDepartmentSpecificRecruitmentProcessForm(
    id: $id
    departmentSpecificRecruitmentProcessForm: $form
  ) {
    id
    department {
      id
    }
    departmentOther
    recruitmentProcessLeadName
    recruitmentProcessLeadJobTitle
    recruitmentProcessLeadEmail
    postingDate
    positions {
      id
      classificationGroup
      classificationLevel
      jobTitle
      employmentTypes
      employmentTypesOther
    }
    advertisementType
    advertisingPlatforms
    advertisingPlatformsOther
    jobAdvertisementLink
  }
}
  ```

```json
{
  "id": "20663acd-9571-4b26-a0dc-c0062f04ae86",
  "form": {
    "department": {
      "connect": "ec817e0d-71ac-4481-9470-d2172f1c58cd"
    },
    "departmentOther": "",
    "recruitmentProcessLeadName": "recruitment_process_lead_name",
    "recruitmentProcessLeadJobTitle": "recruitment_process_lead_job_title",
    "recruitmentProcessLeadEmail": "recruitment@example.org",
    "postingDate": "2025-01-01",
    "positions": {
      "update": [
        {
          "id": "aebc238f-3a3e-422e-92c2-e1ce0c84624b",
          "classificationGroup": "classification_group",
          "classificationLevel": "classification_level",
          "jobTitle": "job_title",
          "employmentTypes": [
            "ASSIGNMENT",
            "OTHER"
          ],
          "employmentTypesOther": "employment_type_other"
        }
      ]
    },
    "advertisementType": "INTERNAL",
    "advertisingPlatforms": [
      "LINKEDIN",
      "OTHER"
    ],
    "advertisingPlatformsOther": "advertising_platforms_other",
    "jobAdvertisementLink": "google.com"
  }
}

```

</details>

<details>
  <summary>DeleteDigitalContractingQuestionnaire</summary>
  
  ```graphql

mutation DeleteDepartmentSpecificRecruitmentProcessForm {
  deleteDepartmentSpecificRecruitmentProcessForm(id: "59d399c9-08ed-4b1f-b10d-f28c6049c450") {
    id
    department {
      id
    }
    departmentOther
    recruitmentProcessLeadName
    recruitmentProcessLeadJobTitle
    recruitmentProcessLeadEmail
    postingDate
    positions {
      id
      classificationGroup
      classificationLevel
      jobTitle
      employmentTypes
      employmentTypesOther
    }
    advertisementType
    advertisingPlatforms
    advertisingPlatformsOther
    jobAdvertisementLink
  }
}

  ```
</details>

### CODE REVIEW

**Comment by petertgiles** [comment]
I just approved #2400 so this one will be going back into in-progress to update the enums once that one is merged.

File: api/app/GraphQL/Validators/DepartmentSpecificRecruitmentProcessPositionBelongsToMany.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    /**
+     * Return the validation rules.
+     *
+     * @return array<string, array<mixed>>
+     */
+    public function rules(): array
+    {
+        return [
+            'create' => ['array', 'min:1'],
```
Comment (esizer) [inline api/app/GraphQL/Validators/DepartmentSpecificRecruitmentProcessPositionBelongsToMany.php:17]: I just have a question about this validation. Right now, if I simply omit the `position` key from variables this rule passes. Is that intended or is position a required field that must have at least one selected?

Comment (esizer) [reply api/app/GraphQL/Validators/DepartmentSpecificRecruitmentProcessPositionBelongsToMany.php:17]: Okay, thanks! Just making sure, based off the issue I wasn't 100% sure if it was required or not 😅

**Review by esizer — APPROVED** [review]
Sweet! 🍭 This looks great to me and all the mutations/validators are working exactly as I expect 👍

File: .vscode/project-words.txt [hunk 2]
```diff
@@ -71,10 +71,12 @@ TELEWORK
 testid
 textbox
 Timeframe
+tiptap
```
[the remark this answers is not in the document]
Comment (petertgiles) [reply .vscode/project-words.txt:74]: Heh, you did.  You're seeing stuff in the merge commit.  :smile:

**Pull request merged**